### PR TITLE
MWPW-147351: Added styles for delayed modal and media block inside

### PR DESCRIPTION
**Pull request by mirafedas**
Added the following changes to the delayed modal styles:
- rounded corners;
- width is respective to the specified in Figma breakpoints.

The modal can contain various fragments with any combination of blocks inside.

When a media block is included in the modal, it will automatically receive the 'in-modal' class.

By default, images within this media block will have padding. For a full-bleed image effect (refer to [Figma](https://www.figma.com/proto/o5PtPudiodDA1mqMnU0TvL/Promo-UX-Enhancements?page-id=0%3A1&type=design&node-id=7-4360&viewport=6113%2C236%2C0.26&t=07APwjmhO5cKwlzC-1&scaling=scale-down-width&starting-point-node-id=38%3A435)), authors should apply the 'full-bleed-image' class to the media block. For a detailed example of such a fragment, see [this document](https://adobe.sharepoint.com/:w:/r/sites/adobecom/_layouts/15/Doc.aspx?sourcedoc=%7B10F5C8C4-4B1D-4BFA-AA27-4EAA1A575E42%7D&file=cc-all-apps-promo-full-bleed-image.docx&action=default&mobileredirect=true).

Resolves: [MWPW-147351](https://jira.corp.adobe.com/browse/MWPW-147351)

See authoring documentation [here](https://main--milo--mirafedas.hlx.live/docs/authoring/promotions/delayed-modal).

**Test URLs:**
- Before:
https://main--cc--adobecom.hlx.page/products/illustrator?instant=2024-05-07T15:00:00.000Z

- After:

1. Image with padding: https://mwpw-147351-delayed-modal-dimensions--milo--mirafedas.hlx.page/drafts/mirafedas/delayed-modals/delayed-modal?martech=off
2. Full-bleed image: https://mwpw-147351-delayed-modal-dimensions--milo--mirafedas.hlx.page/drafts/mirafedas/delayed-modals/delayed-modal-full-bleed-image?martech=off

### CODE REVIEW

**Review by seanchoi-dev — APPROVED** [review]

File: libs/blocks/modal/modal.js [hunk 1]
```diff
@@ -195,6 +207,7 @@ export function getHashParams(hashStr) {
 export function delayedModal(el) {
   const { hash, delay } = getHashParams(el?.dataset.modalHash);
   if (!delay || !hash) return false;
+  isDelayedModal = true;
```
Comment (yesil) [inline libs/blocks/modal/modal.js:210]: should you set `isDelayedModal` to false after closing or better after opening it?

**Review by yesil — APPROVED** [review]

**Comment by Roycethan** [comment]
@mirafedas 
Can you plz review these Ipad differences:

 for Ipad-Pro:
image

for Ipad:
image

**Comment by mirafedas** [comment]
@Roycethan thank you for checking this.
1. The modal close icon in all modals on our website is located inside the modal window, I'm pretty sure it's a mistake on the XD.
2. Fixed, could you please verify?

**Comment by Roycethan** [comment]
> 2\. https://mwpw-147351-delayed-modal-dimensions--milo--mirafedas.hlx.page/drafts/mirafedas/delayed-modals/delayed-modal-full-bleed-image?martech=off

Ok , #2 is fixed.

**Comment by Roycethan** [comment]
#3) Can you plz check this: On mobile: Iphone-Safari - foreground image is not aligned to top leaving strip of background image at the top
image

On Android-chrome looks ok:
image

**Comment by mirafedas** [comment]
@Roycethan 
I'm not able to reproduce it, were there some additional steps done before this span on the top appeared?
Screenshot 2024-05-14 at 12 51 43

**Comment by Roycethan** [comment]
> @Roycethan I'm not able to reproduce it, were there some additional steps done before this span on the top appeared? Screenshot 2024-05-14 at 12 51 43

Its reproducible in IOS-15.2 and not on 15.4

**Comment by mirafedas** [comment]
@Roycethan the reason was that in iOS 15.2 flex gap is not supported, so I rewrote the solution to CSS Grid, now it should be ok, could you please check?

**Comment by Roycethan** [comment]
> @Roycethan the reason was that in iOS 15.2 flex gap is not supported, so I rewrote the solution to CSS Grid, now it should be ok, could you please check?

Fixed thanks

**Pull request merged**